modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -55,14 +55,13 @@ const createParser = function (editor: Editor): DomParser {
 
   // Convert src and href into data-mce-src, data-mce-href and data-mce-style
   parser.addAttributeFilter('src,href,style,tabindex', function (nodes, name) {
-    let i = nodes.length, node;
+    let i = nodes.length, node: Node, value: string;
     const dom = editor.dom;
-    let value, internalName;
+    const internalName = 'data-mce-' + name;
 
     while (i--) {
       node = nodes[i];
-      value = node.attr(name);
-      internalName = 'data-mce-' + name;
+      value = node.attr(name) || '';
```

Comment:
This doesn't look correct, as this will cause the internal attribute to be added when it shouldn't be, as the original attribute no longer exists. Instead, if this value is null or undefined, it should be continuing to the next node.

Also it'd be good to have some tests for this if possible, as it's something that could easily regress?

Reply:
Based on your comments and looking at the code again, I agree that it should just continue to the next node if `value` is undefined. I'll fix that up and see if I can create some tests for it.